ansible/modules/hashivault/hashivault_pki_role_list.py

```diff
@@ -62,8 +62,8 @@ def hashivault_pki_role_list(module):
 
     try:
         return {'data': client.secrets.pki.list_roles(mount_point=mount_point).get('data').get('keys')}
-    except Exception:
-        return {'data': []}
+    except Exception as e:
+        return {'failed': True, 'data': [], 'msg': str(e)}
```

Comment:
Not sure if it's cleaner to drop `data` completely for the `failed` case output or return an empty list?

Reply:
I'm not sure if any other getters support check mode to have a pattern

Reply:
FWIW returning an empty list seems to be consistent with other roles: https://github.com/TerryHowe/ansible-modules-hashivault/blob/main/ansible/modules/hashivault/hashivault_ssh_role_list.py#L80 https://github.com/TerryHowe/ansible-modules-hashivault/blob/main/ansible/modules/hashivault/hashivault_token_role_list.py#L77